scene/resources/animation.cpp

```diff
@@ -815,15 +815,17 @@ int Animation::_insert(float p_time, T &p_keys, const V &p_value) {
 
 	while (true) {
 
-		if (idx == 0 || p_keys[idx - 1].time < p_time) {
-			//condition for insertion.
-			p_keys.insert(idx, p_value);
-			return idx;
-		} else if (p_keys[idx - 1].time == p_time) {
+		// Condition for replacement.
+		if (idx > 0 && Math::abs(p_keys[idx - 1].time - p_time) < CMP_EPSILON) {
```

Comment:
You could use `Math::is_equal_approx()` now (merged recently in master) to simplify this.

Reply:
Thanks, I'll take a look into it.

Reply:
@akien-mga Should be updated to the recommended convention.  In this case it was an approximately zero comparison, but seems that's provided in Math as well.

Reply:
Aren't you comparing `p_keys[idx - 1].time` with `p_time`?

Reply:
Yes, it should be `Math::is_equal_approx(p_keys[idx - 1].time, p_time)`, avoiding these absolute differences comparisons to 0 is the all point of the method :)